numpy/core/src/multiarray/dragon4.c

```diff
@@ -165,18 +215,69 @@ BigInt_Set_uint64(BigInt *i, npy_uint64 val)
     }
 }
 
+#if defined(HAVE_LDOUBLE_IEEE_QUAD_LE)
+static void
+BigInt_Set_2x_uint64(BigInt *i, npy_uint64 hi, npy_uint64 lo)
+{
+    if (hi > bitmask_u64(32)) {
+        i->length = 4;
+    }
+    else if (hi != 0) {
+        i->length = 3;
+    }
+    else if (lo > bitmask_u64(32)) {
+        i->length = 2;
+    }
+    else if (lo != 0) {
+        i->length = 1;
+    }
+    else {
+        i->length = 0;
+    }
+
+    switch (i->length) {
```

Comment:
A comment here like `/* Note deliberate fall through*/` might be nice. It's an unusual construct, but quite clever.